pkg/logs/launchers/integration/launcher_test.go

```diff
@@ -137,17 +177,308 @@ func (suite *LauncherTestSuite) TestIntegrationLogFilePath() {
+		return nil
+	}
+
+	suite.s.Start(nil, nil, nil, nil)
+	suite.integrationsComp.RegisterIntegration(fileID, *mockConf)
+	assert.Equal(suite.T(), 1, len(suite.s.integrationToFile))
+
+	logSample := "hello world"
+	suite.integrationsComp.SendLog(logSample, fileID)
+
+	foundSource := suite.s.sources.GetSources()[0]
+	assert.Equal(suite.T(), foundSource.Config.Type, config.FileType)
+	assert.Equal(suite.T(), foundSource.Config.Source, "foo")
+	assert.Equal(suite.T(), foundSource.Config.Service, "bar")
+
+	assert.Equal(suite.T(), logSample, <-fileLogChan)
+}
+
+// TestSentLogExceedsTotalUsage ensures files are deleted when a sent log causes a
+// disk usage overage
+func (suite *LauncherTestSuite) TestSentLogExceedsTotalUsage() {
+	suite.s.combinedUsageMax = 3 * 1024 * 1024
+
+	filename1 := "sample_integration1_123.log"
+	filename2 := "sample_integration2_123.log"
+	filename3 := "sample_integration3_123.log"
+
+	file1, err := os.Create(filepath.Join(suite.s.runPath, filename1))
+	assert.Nil(suite.T(), err)
+	file2, err := os.Create(filepath.Join(suite.s.runPath, filename2))
+	assert.Nil(suite.T(), err)
+	file3, err := os.Create(filepath.Join(suite.s.runPath, filename3))
+	assert.Nil(suite.T(), err)
+
+	defer func() {
+		file1.Close()
+		file2.Close()
+		file3.Close()
+		os.Remove(filename1)
+		os.Remove(filename2)
+		os.Remove(filename3)
+	}()
+
+	dataOneMB := make([]byte, 1*1024*1024)
+	file1.Write(dataOneMB)
+	file2.Write(dataOneMB)
+	file3.Write(dataOneMB)
+
+	suite.s.Start(nil, nil, nil, nil)
+
+	integrationLog := integrations.IntegrationLog{
+		Log:           "sample log",
+		IntegrationID: "sample_integration1:123",
+	}
+
+	suite.s.receiveLogs(integrationLog)
+
+	file1Stat, err := os.Stat(filepath.Join(suite.s.runPath, filename1))
+	assert.Nil(suite.T(), err)
+	file2Stat, err := os.Stat(filepath.Join(suite.s.runPath, filename2))
+	assert.Nil(suite.T(), err)
+	file3Stat, err := os.Stat(filepath.Join(suite.s.runPath, filename3))
+	assert.Nil(suite.T(), err)
+
+	actualSize := file1Stat.Size() + file2Stat.Size() + file3Stat.Size()
+
+	assert.Equal(suite.T(), suite.s.combinedUsageSize, actualSize)
+	assert.Equal(suite.T(), suite.s.integrationToFile["sample_integration2:123"], suite.s.getLeastRecentlyModifiedFile())
+}
+
+// TestInitialLogsExceedTotalUsageMultipleFiles ensures initial files are deleted if they
+// exceed total allowed disk space
+func (suite *LauncherTestSuite) TestInitialLogsExceedTotalUsageMultipleFiles() {
+	oneMB := int64(1 * 1024 * 1024)
+	suite.s.combinedUsageMax = oneMB
+
+	filename1 := "sample_integration1_123.log"
+	filename2 := "sample_integration2_123.log"
+
+	dataOneMB := make([]byte, oneMB)
+
+	file1, err := os.Create(filepath.Join(suite.s.runPath, filename1))
+	assert.Nil(suite.T(), err)
+	file2, err := os.Create(filepath.Join(suite.s.runPath, filename2))
+	assert.Nil(suite.T(), err)
+
+	defer func() {
+		file1.Close()
+		file2.Close()
+		os.Remove(filename1)
+		os.Remove(filename2)
+	}()
+
+	file1.Write(dataOneMB)
```

Comment:
Missing on a few places where you write to the file a call to `Flush`.
```suggestion
	file1.Write(dataOneMB)
	_ = file1.Flush()
```

Reply:
The problem with windows tests was files were scheduled to be deleted before calling `file.Close()`.